src/shared_modules/content_manager/tests/unit/CtiDownloader_test.cpp

```diff
@@ -250,7 +224,205 @@ TEST_F(CtiDownloaderTest, BaseParametersDownloadInterrupted)
     EXPECT_EQ(m_spUpdaterContext->data, expectedData);
 
     // Check expected base parameters.
-    EXPECT_EQ(downloader.getLastOffset(), DEFAULT_LAST_OFFSET);
-    EXPECT_EQ(downloader.getLastSnapshotLink(), DEFAULT_LAST_SNAPSHOT_LINK);
-    EXPECT_EQ(downloader.getLastSnapshotOffset(), DEFAULT_LAST_OFFSET);
+    const auto parameters {downloader.getParameters()};
+    EXPECT_FALSE(parameters->lastOffset.has_value());
+    EXPECT_FALSE(parameters->lastSnapshotLink.has_value());
+    EXPECT_FALSE(parameters->lastSnapshotOffset.has_value());
+}
+
+/**
+ * @brief Tests the download of metadata with invalid JSON format.
+ *
+ */
+TEST_F(CtiDownloaderTest, BaseParametersDownloadMetadataInvalidFormat)
+{
+    auto mockMetadata = R"({data":{})";
+    m_spFakeServer->setCtiMetadata(std::move(mockMetadata));
+
+    auto downloader {CtiDummyDownloader(HTTPRequest::instance())};
+    ASSERT_THROW(downloader.handleRequest(m_spUpdaterContext), std::runtime_error);
+
+    // Check expected data.
+    nlohmann::json expectedData;
+    expectedData["paths"] = m_spUpdaterContext->data.at("paths");
+    expectedData["stageStatus"] = FAIL_STATUS;
+    expectedData["type"] = CONTENT_TYPE;
+    expectedData["offset"] = 0;
+    EXPECT_EQ(m_spUpdaterContext->data, expectedData);
+}
+
+/**
+ * @brief Tests the download of metadata with missing last_snapshot_offset key.
+ *
+ */
+TEST_F(CtiDownloaderTest, BaseParametersDownloadMetadataMissingLastSnapshotOffsetKey)
+{
+    auto mockMetadata = R"(
+        {
+            "data":
+            {
+                "ignored_key": true,
+                "last_offset": 100,
+                "last_snapshot_link": "some_link"
+            }
+        }
+    )"_json.dump();
```

Comment:
This could be directly a string:
```suggestion
    std::string mockMetadata = R"(
        {
            "data":
            {
                "ignored_key": true,
                "last_offset": 100,
                "last_snapshot_link": "some_link"
            }
        }
    )";
```
Also on the other new tests.
(I used std::string instead of auto to keep it 'movable')